tests/e2e/workloads/bdi/fixtures.py

```diff
@@ -0,0 +1,253 @@
+import pytest
+import os
+import logging
+from ocs_ci.utility.utils import run_cmd, clone_repo, exec_cmd
+from ocs_ci.ocs import machine, node, ocp, constants
+from ocs_ci.framework import config
+import tempfile
+from ocs_ci.utility import templating
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def clone_ibm_chart_project(request):
```

Comment:
This should be a class without fixtures and use the init values from what you have defined in the test, the init should be the values that you have defined in the test case test_bdi_workload.py , you should check pgsql class to get some idea and how we initialize it across multiple tests.

Reply:
Hi @vasukulkarni , could you please elaborate on this request?
Why the ask to avoid the usage of fixtures?

Reply:
@RazTamir I am asking to follow the pattern we have for the pgsql,amq, ripsaw etc - https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/amq.py ,  we can use fixtures outside not for the library itself like we are doing for all other cases.

Reply:
Thanks for that.
I have to say the amq design is wrong:
1) the setup is happening here https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/amq.py#L871
2) in case all goes well, we are setting the flag amq_is_setup to True - https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/amq.py#L888
3) in the teardown we are checking this flag and performing the teardown based on its value - https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/amq.py#L906

What will happen if the setup failed during one of the calls to the functions?
You will have leftovers as the flag amq_is_setup will never be set to True.

This is exactly what the fixtures come to help you with.
I honestly don't see a reason to change the existing PR (related to the design of setup/teardown) as we are following the right methodology

Reply:
Thanks Raz. Looks like teardown part in AMQ should be improved to handle such cases.
@Akarsha-rai FYI

Reply:
@RazTamir Sorry I missed this one, afaik there is nothing wrong with the design, we are using the same test across so many places and they are running fine, the functions are correctly used in the *fixtures* setup and teardown as seen here in one of the test case https://github.com/red-hat-storage/ocs-ci/blob/master/tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py#L64-L70 

maybe there are some small improvements but the teardown deletes name space https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/amq.py#L937-L941 and since its used in fixture its *guaranteed* to cleanup

We use setup/teardown in fixtures ( initialized according to what test requires) so the assumption we are not using fixtures is not correct.

We also documented how to avoid improper fixture usages here - https://ocs-ci.readthedocs.io/en/latest/fixture_usage.html#avoid  and the current PR is following one of the old procedure.

I can setup a meeting to go over the confusion.